up-l3/usubscription/v3/README.adoc

```diff
@@ -18,139 +18,615 @@ SPDX-FileType: DOCUMENTATION
+--
+
+[.specitem,oft-sid="dsn~usubscription-fetch-subscriptions-invalid-topic~1",oft-needs="impl,utest"]
+--
+When receiving a `FetchSubscriptions()` request that contains a topic that
+* is not a valid xref:../../../basics/uri.adoc[uProtocol URI] or
+* contains a _wildcard_ authority or
+* contains a _wildcard_ uEntity ID (`ue_id`) or
+* contains a _wildcard_ resource ID,
+
+a uSubscription service *MUST* return a failure status message with link:../../../up-core-api/uprotocol/v1/ucode.proto[`UCode`] `INVALID_ARGUMENT`.
+--
+
+[.specitem,oft-sid="dsn~usubscription-fetch-subscriptions-invalid-subscriber~1",oft-needs="impl,utest"]
+--
+When receiving a `FetchSubscriptions()` request that contains a subscriber URI that
+* is not a valid xref:../../../basics/uri.adoc[uProtocol URI] or
+* contains a _wildcard_ authority or
+* contains a _wildcard_ uEntity ID (`ue_id`) or
+* contains a _wildcard_ resource ID,
+
+a uSubscription service *MUST* return a failure status message with link:../../../up-core-api/uprotocol/v1/ucode.proto[`UCode`] `INVALID_ARGUMENT`.
+--
+
+[.specitem,oft-sid="req~usubscription-fetch-subscriptions-has-more-records~1",oft-needs="impl,utest"]
+--
+If the list of subscriptions returned in response to a `FetchSubscriptions()` call does not contain all existing subscription relationships, uSubscription service *MUST* set the `has_more_records` field of the `FetchSubscriptionsResponse` to `true`.
+--
+
+[.specitem,oft-sid="req~usubscription-fetch-subscriptions-stable-sorting~1",oft-needs="impl,utest"]
+--
+Subscription entries returned by subsequent calls to the `FetchSubscriptions()` function *MUST* be in identical order.
+--
+
+[.specitem,oft-sid="req~usubscription-fetch-subscriptions-has-more-records~1",oft-needs="impl,utest"]
+--
+If the list of subscriptions returned in response to a `FetchSubscriptions()` call does not contain all existing subscription relationships, uSubscription service *MUST* set the `has_more_records` field of the `FetchSubscriptionsResponse` to `true`.
+--
+
+[.specitem,oft-sid="req~usubscription-fetch-subscriptions-offset~1",oft-needs="impl,utest"]
+--
+When a client calls the `FetchSubscriptions()` function with an `offset` argument, the list of subscriptions returned by uSubscription service *MUST* omit all subscription relationship entries up to the provided offset.
+--
+
+[#usubscription-topic]
 == Topics
 
-A topic describes what the calling applications (subscribers) wants to subscribe too. Topic are expressed in link:../../../basics/README.adoc#_uprotocol_uri[uProtocol URIs] format. Topics used in uSubscription APIs have the additional requirements of:
+A topic identifies the message resource that a link:#usubscription-subscriber[Subscriber] wants to subscribe to. Topic are expressed in xref:../../../basics/uri.adoc[uProtocol URI] format.
```

Comment:
Topic section probably should be at the beginning under overview or somewhere else as it is fundamental to the spec

Reply:
ok